### PR TITLE
feat!: remove legacy client run-examples command

### PR DESCRIPTION
**Pull request by blgm**
- New "csb run-examples" and "csb examples" commands have been introduced and we have got good feedback on them
- This commit removes the "csb client run-examples" and "csb client examples" commands since we believe they are now superfluous
- The commands can be added back by setting the feature flag `CSB_ENABLE_LEGACY_EXAMPLES_COMMANDS=1`, but if you do this, please raise an issue to let us know as otherwise we will remove the functionality completely in a future release

### CODE REVIEW

File: docs/brokerpak-intro.md [hunk 1]
```diff
@@ -227,34 +227,20 @@ If the *examples* section of the brokerpak is not empty, it is possible (and adv
...
-
-```bash
-docker run --rm -v ${BROKERPAK_SRC_DIR}:/brokerpak -w /brokerpak \
--e "SECURITY_USER_NAME=csb-un" \
--e "SECURITY_USER_PASSWORD=csb-pw" \
--e "GSB_API_HOSTNAME=host.docker.internal" \
--e "GSB_API_PORT=8080" \
--e USER \
-cfplatformeng/csb pak run-examples /brokerpak/$(ls *.brokerpak)
+cfplatformeng/csb run-examples --all
```
Comment (zucchinidev) [inline docs/brokerpak-intro.md:243]: praise: good catch

File: pkg/featureflags/feature_flags.go [hunk 2]
```diff
@@ -9,13 +9,20 @@ const (
...
+	// the old way to stay around.
+	EnableLegacyExamplesCommands FeatureFlagName = "legacy.examples.enabled"
 )
 
 func init() {
 	for ffName, varName := range map[FeatureFlagName]string{
-		TfUpgradeEnabled:                 "TERRAFORM_UPGRADES_ENABLED",
-		DynamicHCLEnabled:                "BROKERPAK_UPDATES_ENABLED",
+		TfUpgradeEnabled:                 "TERRAFORM_UPGRADES_ENABLED", // deprecated pattern - future variables should start CSB_
+		DynamicHCLEnabled:                "BROKERPAK_UPDATES_ENABLED",  // deprecated pattern - future variables should start CSB_
```
Comment (zucchinidev) [inline pkg/featureflags/feature_flags.go:23]: praise: great idea. I like the pattern

File: cmd/client.go [hunk 3]
```diff
@@ -150,7 +150,10 @@ user-defined plans.
 		},
 	}
 
-	clientCmd.AddCommand(clientCatalogCmd, provisionCmd, deprovisionCmd, bindCmd, unbindCmd, lastCmd, runExamplesCmd, updateCmd, examplesCmd)
+	clientCmd.AddCommand(clientCatalogCmd, provisionCmd, deprovisionCmd, bindCmd, unbindCmd, lastCmd, updateCmd)
+	if featureflags.Enabled(featureflags.EnableLegacyExamplesCommands) {
```
Comment (zucchinidev) [inline cmd/client.go:154]: question: this feature flag has to be defined in the pipeline?

**Review by zucchinidev — APPROVED** [review]

File: cmd/client.go (hunk 3 above)
Comment (blgm) [reply cmd/client.go:154]: No. The CI pipeline uses the new way to run examples.

Comment (zucchinidev) [reply cmd/client.go:154]: Thanks!!

**Pull request merged**